onnx/defs/shape_inference.h

```diff
@@ -88,6 +88,8 @@ struct InferenceContext {
   virtual GraphInferencer* getGraphAttributeInferencer(const std::string& attribute_name) = 0;
   virtual ~InferenceContext() {}
   virtual const SparseTensorProto* getInputSparseData(size_t index) const = 0;
+  // Gets the shape inputs computed by partial data propagation.
+  virtual const TensorShapeProto* getShapeInput(size_t index) const = 0;
```

Comment:
`getShapeInput` here in InferenceContext is basically same as `TensorShapeProto* getInputData` in DataPropagationContext, right?

Actually I am thinking that whether to change `getInputData` into `getShapeInput` in DataPropagationContext to prevent confusion. Otherwise, `getInputData` functions in InferenceContext and DataPropagationContext are doing different things...

Reply:
I would recommend ```getSymbolicInput``` to avoid confusion.

Reply:
Logically, this subsumes ```getInputData```. Eg., in the long term, we could conceivably replace ```getInputData``` by this function (just like in DataPropagationContext), but that would require updating all current uses. If possible, I would like to avoid a potential confusion between ```Get input data encoded as a TensorShapeProto``` (which is what this is) and ```Get shape of input data``` (which this is not).